src/vcpkg/dependencies.cpp

```diff
@@ -593,62 +592,59 @@ namespace vcpkg
             return {};
     }
 
-    std::vector<RemovePlanAction> create_remove_plan(const std::vector<PackageSpec>& specs,
-                                                     const StatusParagraphs& status_db)
+    RemovePlan create_remove_plan(const std::vector<PackageSpec>& specs, const StatusParagraphs& status_db)
     {
-        struct RemoveAdjacencyProvider final : AdjacencyProvider<PackageSpec, RemovePlanAction>
+        struct RemoveAdjacencyProvider final : AdjacencyProvider<PackageSpec, PackageSpec>
         {
-            const StatusParagraphs& status_db;
-            const std::vector<InstalledPackageView>& installed_ports;
-            const std::unordered_set<PackageSpec>& specs_as_set;
+            std::unordered_map<PackageSpec, std::vector<PackageSpec>> rev_edges;
 
-            RemoveAdjacencyProvider(const StatusParagraphs& status_db,
-                                    const std::vector<InstalledPackageView>& installed_ports,
-                                    const std::unordered_set<PackageSpec>& specs_as_set)
-                : status_db(status_db), installed_ports(installed_ports), specs_as_set(specs_as_set)
+            std::vector<PackageSpec> adjacency_list(const PackageSpec& spec) const override
```

Comment:
As an aside, should this be:
```suggestion
            const std::vector<PackageSpec>& adjacency_list(const PackageSpec& spec) const override
```

or do we know a copy must always be made here?

No change requested.

Reply:
The Graph resolver doesn't know that there _won't_ be a copy, so the interface is pessimistic.